### PR TITLE
feat(connector): [Wordline] Use connector_response_reference_id as reference to merchant

### LINKED ISSUE #2351: [FEATURE]: [Worldline] Use `connector_response_reference_id` as reference to merchant
### :memo: Feature Description

- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as `payment_id`, `attempt_id`, and `connector_transaction_id` for a single transaction. 
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

### :hammer: Possible Implementation

- When we receive a response from the connector for the payment, we deserialize it and populate the `response` field in RouterData. 
- For the `TransactionResponse` type, we must fill the `connector_response_reference_id` with a corresponding reference id for the merchant to identify the transaction. 
- One might need to have exposure to api docs of the connector for which it is being implemented to decide what to fill in connector_response_reference_id.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1735

🔖 Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by Kota-Karthik**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

The changes I made are:-
image

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Closes #2351.

## How did you test it?

To address potential lints in the code, I  considered running the command:
cargo clippy --all-features
Additionally, to ensure consistent code formatting, I used:
cargo +nightly fmt
these encountered any errors during these processes. This diligence will contribute to a smoother workflow and help prevent workflow failures.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

I had solved the issue #2351 assigned to me within the given time 
kindly review and merge into the main branch

### CODE REVIEW

**Comment by Kota-Karthik** [comment]
@AkshayaFoiger 
how should i name the PR?

**Comment by AkshayaFoiger** [comment]
> @AkshayaFoiger how should i name the PR?

Hey @Kota-Karthik , to ensure proper naming conventions for the pull request (PR), kindly consult the guidelines outlined in the [CONTRIBUTING.md](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md) and observe the example provided in PR #1735

**Comment by Kota-Karthik** [comment]
@AkshayaFoiger I had checked the contributing guidelines as well as the reference PR #1735 you told to check ,
I had named my PR according to it 
But still my workflow tests is failing
kindly explain what,s wrong with my PR title
Thank You : )

**Comment by AkshayaFoiger** [comment]
Hello @Kota-Karthik,

I wanted to bring to your attention that, according to our contribution documentation, we typically use the following syntax: `<type>(<scope>): <short summary>` for PR titles and we don't have a type called **Feature** in our conventions.

This difference in naming might be the reason for the workflow check failure. Would you please consider renaming the PR to:

Title: feat(connector): [Wordline] Add connector response reference ID support.

I believe this adjustment would help the PR align with our established guidelines.

**Comment by Kota-Karthik** [comment]
@AkshayaFoiger 
My PR is passing 5 tests and failing others 
could you explain what should be done to make all tests pass?

**Comment by AkshayaFoiger** [comment]
This PR addresses [Issue #2351](https://github.com/juspay/hyperswitch/issues/2351).

**Comment by Kota-Karthik** [comment]
@AkshayaFoiger 
yes it addresses issue #2351 and i have mentioned it in the PR

**Comment by AkshayaFoiger** [comment]
Hello @Kota-Karthik ,

I would like to kindly request you to refer [Cargo Commands](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md#cargo-commands) mentioned in contribution doc.

As mentioned in it, to address potential lints in the code, please consider running the command:

`cargo clippy --all-features`

Additionally, to ensure consistent code formatting, you may run:

`cargo +nightly fmt`

Should you encounter any errors during these processes, I encourage you to take the necessary steps to rectify them. This diligence will contribute to a smoother workflow and help prevent workflow failures.

**Comment by Kota-Karthik** [comment]
@AkshayaFoiger 
please approve the workflows
so that i can complete this PR and move on to next issue!
Thank You : )

File: crates/router/src/connector/worldline/transformers.rs [hunk 1]
```diff
@@ -174,6 +174,7 @@ pub struct PaymentsRequest {
     pub payment_data: WorldlinePaymentMethod,
     pub order: Order,
     pub shipping: Option<Shipping>,
+    pub reference: String,
```
Comment (srujanchikke) [inline crates/router/src/connector/worldline/transformers.rs:177]: Hello @Kota-Karthik ,

In reference to issue #2351, we need to adjust the reference field mapping based on the API documentation of the connector. The merchant reference ID should be structured as follows: `order` -> `references` -> `merchantReference` for the request . For the response, `connector_response_reference_id` should be mapped to  `paymentOutput` -> `references` -> `merchantReference`. Kindly make the required modifications based on their documentation. Additionally, there were some unnecessary files pushed in your recent commits; could you please remove them?

You can find the reference for Worldline API documentation [here](https://apireference.connect.worldline-solutions.com/s2sapi/v1/en_US/java/payments/create.html?paymentPlatform=ALL).

Thank you.

**Comment by VedantKhairnar** [comment]
Hey @Kota-Karthik ,
Let us know if you face any issues. Happy to help! : )

**Comment by VedantKhairnar** [comment]
Hey @Kota-Karthik ,
Just checking in - are you working on it or shall we consider reassigning the issue to someone else?

**Pull request closed without merging**